src/Tempest/Container/InitializerDiscovery.php

```diff
@@ -8,12 +8,15 @@
 use Tempest\Discovery\HandlesDiscoveryCache;
 use Tempest\Support\Reflection\ClassReflector;
 
+/**
+ * @property GenericContainer $container
+ */
 final readonly class InitializerDiscovery implements Discovery
 {
     use HandlesDiscoveryCache;
 
     public function __construct(
-        private Container&GenericContainer $container,
+        private Container $container,
```

Comment:
simplified to only the interface, and type hint to help the IDE see what methods are actually available.

[rector]